doc/releases/migration-guide-4.2.rst

```diff
@@ -306,6 +306,16 @@ SPI
 * Renamed the device tree property ``port_sel`` to ``port-sel``.
 * Renamed the device tree property ``chip_select`` to ``chip-select``.
 
+xSPI
+===
+
+* For STM32, the external memory base address and max possible size was given by the reg property
+  of the compatible = "st,stm32-xspi"; (usually in the device DTS).
+  The reg property of the compatible = "st,stm32-xspi-nor"; (usually in the board DTS)
+  is now given <0> and the <size> property gives the actual size of the memory device in bits.
```

Comment:
Let's be a bit more specific:

* On STM32 devices, external memories device tree descriptions for size and address are now split in two separate
  properties to comply with specification recommendations.
  For instance, following external flash description
      reg = <0x70000000 DT_SIZE_M(64)>; /* 512 Mbits */
  is changed to
      reg = <0>;
		size = <DT_SIZE_M(512)>; /* 512 Mbits */
 Note that the <size> property gives the actual size of the memory device in bits.
 Previous mapping address information is now described in xspi node at SoC dtsi level.